doc/HWReleaseProcess.md

```diff
@@ -19,10 +19,11 @@
 
 ## Major Releases
 
-1. The RTL team will consume the **release collateral** generated by the **Nightly Workflow** and verify it in the DV environment. The verification should include RTL synthesis using the chosen tool from the Caliptra WG.
-2. Upon success of the DV verification a new release will be made on the **caliptra** repo
-    -  A ZIP file containing the release collateral will be included as part of the release.
-4. The **caliptra-rtl** and **caliptra-sw** repos will be tagged with a **release** version tag.
+1. Before finalizing an RTL release, the hardware team will follow the procedure described in [CaliptraReleaseChecklist.md](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraReleaseChecklist.md).
+  - Upon finalizing all release checklist items, the hardware team will generate a release tag on the RTL **main** branch, using the following naming convention: `v<major>.<minor>`.
+  - When the checklist items are completed, the RTL team will create a new branch following the naming convention `patch_v<major>.<minor>`. This branch will exactly match the contents of the major release tag. Any future fixes that must be applied as patches to this release version will be committed to the `patch_v<major>.<minor>` branch. Branch **main** will continue to receive updated code for the subsequent planned release, and is not guaranteed to contain a valid release except as tagged.
+2. The software team will follow the [submodule update steps](#RTL-Submodule-Pointer-Update) to revise the RTL submodule pointer to consume the code from branch `patch_v<major>.<minor>` instead of from branch **main**. 
```

Comment:
Overall this makes perfect sense for older versions. But in the scenario for a version that is in the earlier stages of development, there will be a need to consume RTL before it's officially released. This doesn't really allow for that unless pre-releases are also committed to these patch branches.

Reply:
The submodule at hw/latest is meant to track the RTL main branch. We have hw/1.0 and we can have hw/1.1 track those associated RTL patch branches.
When the SW repo creates a release branch, latest might need to be re-synchronized to point to the correct RTL branch though.

Reply:
The intent here is basically what @jlmahowa-amd suggests -- SW can track the tip of main (or patch_vX.Y) during development, but switch to using the Git Tag when cutting a release.

Reply:
I'm glad we are on the same page but I don't think this step in the process says the same thing.
From inspection I _believe_ that the automatic RTL update also covers the hw/1.0 branch.
Maybe we should add creating a hw/<major>.<minor> submodule to the SW release process and make that submodule be the one associated with the release. Nick tells me that we tag releases and don't necessarily make a branch.
Here is my suggestion, keeping in mind that this changes one of my key assumptions about how the hw/latest is the RTL associated with the SW release (and we would probably need to be more explicit about what RTL is associated with the SW release).
(Bold to show changes)
2. The software team will follow the [submodule update steps](#RTL-Submodule-Pointer-Update) to **create a hw/<major>.<minor>** RTL submodule pointer to consume the code from branch `patch_v<major>.<minor>` **and associate that submodule with the release**.

Reply:
The key takeaway here is that RTL/main will continue to add new features for subsequent releases after the tag is applied for a release. This means SW must consume RTL at the tag (or patch branch) rather than from main while continuing development towards a SW release of the same version.

I can make your modification, as I think it accounts for this. But for clarification, why would SW team want multiple submodules, rather than just using the latest and greatest? Is the expectation that retroactive patch releases will be cut from the tip of main, just using the appropriate submodule (which sounds risky to me)?

Reply:
> But for clarification, why would SW team want multiple submodules, rather than just using the latest and greatest? 

We would like our runtime firmware source code to be compatible with old hardware. For example, we currently have a hw/1.0 submodule that we use for regression testing the latest runtime firmware with the 1.0 hardware, to make sure it compiles and the tests pass.

It's not practical to regression test with EVERY hardware release, but we should at least try to get representation from each "minor" branch (1.0.x, 1.1.x, etc).

Reply:
FWIW, due to https://github.com/chipsalliance/caliptra-sw/issues/1410, I'm not convinced we should be "releasing" anything at all, instead simply telling our customers which git tags they should be building from.

Reply:
Thanks for explaining -- I pushed the update suggested by @jlmahowa-amd.